src/main/java/org/datadog/jmxfetch/util/JeeStatisticsAttributes.java

```diff
@@ -0,0 +1,328 @@
+package org.datadog.jmxfetch.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.sql.Ref;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.WeakHashMap;
+import javax.management.AttributeNotFoundException;
+import javax.management.ReflectionException;
+
+public class JeeStatisticsAttributes {
+    /** Attributes for @see javax.management.j2ee.statistics.CountStatistic */
+    private static final List<String> COUNT_ATTRIBUTES = Collections.singletonList("count");
+
+    /** Attributes for @see javax.management.j2ee.statistics.BoundaryStatistic */
+    private static final List<String> BOUNDARY_ATTRIBUTES =
+        Collections.unmodifiableList(Arrays.asList("upperBound", "lowerBound"));
+
+    /** Attributes for @see javax.management.j2ee.statistics.TimeStatistic */
+    private static final List<String> TIME_ATTRIBUTES =
+        Collections.unmodifiableList(Arrays.asList("count", "minTime", "maxTime", "totalTime"));
+
+    /** Attributes for @see javax.management.j2ee.statistics.RangeStatistic */
+    private static final List<String> RANGE_ATTRIBUTES =
+        Collections.unmodifiableList(Arrays.asList("highWaterMark", "lowWaterMark", "current"));
+
+    /** Attributes for @see javax.management.j2ee.statistics.BoundedRangeStatistic */
+    private static final List<String> BOUNDED_RANGE_ATTRIBUTES =
+        Collections.unmodifiableList(
+            Arrays.asList("upperBound", "lowerBound", "highWaterMark", "lowWaterMark", "current"));
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JeeStatisticsAttributes.class);
+    private static final WeakHashMap<ClassLoader, ReflectionHolder> REFLECTION_CACHE =
+        new WeakHashMap<>();
```

Comment:
I suspect this won't work as intended in terms of unloading, because the values in the map contain class references from the class-loader, and those will stop the class-loader from being unloaded. We could wrap the values in a `SoftReference` to avoid pinning the class-loader while allowing the value to carry over across GC cycles.

Reply:
Wrapped around a softreference. Thank you for the suggestion. I completely missed this